iothub_client/src/iothubtransport_mqtt_common.c

```diff
@@ -2460,7 +2472,8 @@ static void ProcessPendingTelemetryMessages(PMQTTTRANSPORT_HANDLE_DATA transport
 
         if (((current_ms - msg_detail_entry->msgPublishTime) / 1000) > RESEND_TIMEOUT_VALUE_MIN)
         {
-            if (msg_detail_entry->retryCount >= MAX_SEND_RECOUNT_LIMIT)
+            // if message has been retried too many times normally, or too many times on reconnect (and this is a reconnect), expire the message
+            if (msg_detail_entry->retryCount >= MAX_SEND_RECOUNT_LIMIT || (msg_detail_entry->isReconnectRetry && msg_detail_entry->reconnectRetryCount >= MAX_SEND_ON_RECONNECT_LIMIT))
```

Comment:
probably ok with just

if (msg_detail_entry->retryCount >= MAX_SEND_RECOUNT_LIMIT || msg_detail_entry->reconnectRetryCount >= MAX_SEND_ON_RECONNECT_LIMIT)

The flag was used to figure out which one to increment below